frontend/src/ui/Input.tsx

```diff
@@ -50,6 +50,93 @@ export const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
     ),
 );
 
+type InputWithCheckboxProps = {
+    checkboxChecked: boolean;
+    setCheckboxChecked: (newValue: boolean) => void;
+    label: string;
+    input: ReactNode;
+}
+
+/** Checkbox with a label to enable/disable an adjacent input */
+export const InputWithCheckbox: React.FC<InputWithCheckboxProps> = (
+    { checkboxChecked, setCheckboxChecked, label, input }
+) => <div css={{ display: "flex", flexDirection: "row", alignItems: "center" }}>
+    <input
+        type="checkbox"
+        checked={checkboxChecked}
+        onChange={() => setCheckboxChecked(!checkboxChecked)}
+        css={{ margin: "0 4px" }}
+    />
+    <label css={{ color: COLORS.neutral90, fontSize: 14 }}>
+        {label}
+    </label>
+    {input}
+</div>;
+
+type TimeInputProps = {
+    timestamp: string;
+    setTimestamp: (newTime: string) => void;
+    disabled: boolean;
+}
+
+/** A custom three-part input for time inputs split into hours, minutes and seconds */
+export const TimeInput: React.FC<TimeInputProps> = ({ timestamp, setTimestamp, disabled }) => {
+    const timeParts = (/(\d+h)?(\d+m)?(\d+s)?/).exec(timestamp)?.slice(1) ?? [];
+    const [hours, minutes, seconds] = timeParts
+        .map(part => part ? parseInt(part.replace(/\D/g, "")) : 0);
+
+    const handleTimeChange = (newValue: number, type: TimeUnit) => {
+        if (isNaN(newValue)) {
+            return;
+        }
+
+        const cappedValue = Math.min(newValue, 59);
+        const newTimestamp = `${type === "h" ? cappedValue : hours}h`
+            + `${type === "m" ? cappedValue : minutes}m`
+            + `${type === "s" ? cappedValue : seconds}s`;
+
+        setTimestamp(newTimestamp);
+    };
+
+    type TimeUnit = "h" | "m" | "s";
+    const entries: [number, TimeUnit][] = [
+        [hours, "h"],
+        [minutes, "m"],
+        [seconds, "s"],
+    ];
+
+    return (
+        <div css={{ color: disabled ? COLORS.neutral70 : COLORS.neutral90 }}>
+            {entries.map(([time, unit]) => <Fragment key={`${unit}-input`}>
+                <input
```

Comment:
> Ideally ideally, the "hour" and "minute" inputs will only be shown if the video is at least one hour/one minute.

Agreed, and this is doable on the video page with the paella api, though afaik only once the video is loaded (i.e. once the user presses play). This does unfortunately mean that we can't do that on the "video details" page.

It would be nice if we could use the event's `duration` value instead, but as noted in #884 that isn't always correct, so we can't.

Reply:
Oh mh good point. Then let's not do the duration-dependent stuff at all. We can revisit once #884 is done.

Reply:
> Now that we use three inputs, you can use type="number" here, right? You can then just set the max and min attributes to get built-in validation. That would also solve the problem with the -3.

Regarding this: I did not use `type="number"` since then that comes with additional input elements... which can be removed but also need browser specific code which isn't too nice. I think dealing with the input validation inside the input handler is fine.